src/Orchard.Web/Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -60,9 +60,9 @@ public class AutoroutePartHandler : ContentHandler {
             // Should it become the home page?
             if (part.PromoteToHomePage) {
                 // Get the current homepage an unmark it as the homepage.
-                var currentHomePage = _homeAliasService.GetHomePage(VersionOptions.Latest);
-                if(currentHomePage != null && currentHomePage.Id != part.Id) {
-                    var autoroutePart = currentHomePage.As<AutoroutePart>();
+                var currentHomePageId = _homeAliasService.GetHomePageId();
+                if(currentHomePageId != part.Id) {
+                    var autoroutePart = part.As<AutoroutePart>();
```

Comment:
This changes completely what the handler is doing. The live version is removing the `PromoteToHomePage` falg from the current homepage, rather than from the AutoroutePart being handled right now.

Reply:
I will make more tests but I've not had issues with editing/publishing/creating a home page.